_posts/2022-04-07-how-to-use-kerberos-in-copr.md

```diff
@@ -0,0 +1,84 @@
+---
+title: How to use Kerberos in Copr
+author: Silvie Chlupova
+layout: post
+
+
+During the last release, a new authentication option was added to Copr. Now users can also log in using a kerberos ticket.
+We use GSSAPI to implement this mechanism.
+The GSSAPI (Generic Security Services API) is a common interface for accessing the Kerberos security system.
+Enabling GSSAPI authentication in Copr CLI or API will allow authentication without using  the API token for clients who have a Fedora account.
+To use GSSAPI login in WebUI click on `gssapi-login`  (negotiate needs to be allowed for the .fedorainfracloud.org domain in the browser, [see browser configuration](https://fedoraproject.org/wiki/Infrastructure/Kerberos#Firefox)).
+It will log you in if you have a kerberos ticket for `username@FEDORAPROJECT.ORG`.
+To enable GSSAPI you need to obtain a Kerberos ticket first. If you want to work with Fedora Copr, you can just do:
+
+```shell
+$ fkinit # optionally with "-u username"
+Enter your password and OTP concatenated. (Ignore that the prompt is for only the token)
+Enter OTP Token Value: <your password + OTP token>
+$ copr whoami
+your_username
+```
+You can also find here [how to use kerberos auth with Fedora Infrastructure](https://fedoraproject.org/wiki/Infrastructure/Kerberos).
+To work with another (non-Fedora Copr) instance, you need to obtain the kerberos ticket differently, and you need to have a configuration file containing the appropriate copr_url:
+
+```shell
+$ kinit username@EXAMPLE.COM
+$ cat ~/.config/copr
+[copr-cli]
+copr_url = https://copr.example.com/
+```
+
+## API key vs. Kerberos
+If you have an API key set up, and you also have a kerberos ticket, then the API key has a higher priority and will be used.
```

Comment:
If you have an API key set up and have a Kerberos ticket, then the API key has a higher priority and will be used.